core/commands/pin.go

```diff
@@ -475,72 +477,40 @@ func pinLsKeys(req *cmds.Request, typeStr string, n *core.IpfsNode, api coreifac
 	return nil
 }
 
-func pinLsAll(req *cmds.Request, typeStr string, n *core.IpfsNode, emit func(value interface{}) error) error {
+func pinLsAll(req *cmds.Request, typeStr string, pinning pin.Pinner, dag ipld.DAGService, emit func(value interface{}) error) error {
+	pinCh, errCh := coreapi.PinLsAll(req.Context, typeStr, pinning, dag)
+
 	enc, err := cmdenv.GetCidEncoder(req)
 	if err != nil {
 		return err
 	}
 
-	keys := cid.NewSet()
-
-	AddToResultKeys := func(keyList []cid.Cid, typeStr string) error {
-		for _, c := range keyList {
-			if keys.Visit(c) {
-				err := emit(&PinLsOutputWrapper{
-					PinLsObject: PinLsObject{
-						Type: typeStr,
-						Cid:  enc.Encode(c),
-					},
-				})
-				if err != nil {
-					return err
-				}
+	procErr := make(chan error, 1)
+	ctx := req.Context
+	for {
+		select {
+		case p, ok := <-pinCh:
+			if !ok {
+				procErr <- nil
```

Comment:
Unnecessary channel?

Reply:
yep, I think a mistake from a some git mistakes. thx for the catch.